### PR TITLE
Add sequentially growable transformer backbones and classifiers

### PR DESCRIPTION
**Pull request by Santiago23florido**
## Summary

This PR adds transformer-based growable models to the codebase and integrates them with the sequential growth workflow.

## What’s included

- Add growable transformer architectures and supporting components
- Integrate transformer containers with `SequentialGrowingModel`
- Support selecting a specific transformer block for growth with `set_growing_layers(index=...)`
- Support sequential block scheduling for transformer growth
- Delegate growth helper methods from transformer blocks to their growable MLP branches
- Add tests covering sequential transformer growth behavior

## Notes
Derived image models such as CCT, ViT-Lite, and CVT inherit that behavior through `GrowingTransformer`.

Consideration of Modifications in the future  order to create function preserving skip connections that allow to add blocks in the structure

| Model | Final test accuracy | Number of parameters |
|---|---:|---:|
| `cct_7_3x1_fogro` | 0.8903 | 2,938,313 |
| `cct_7_3x1_base` | 0.9187 | 3,760,139 |
| `cct_7_3x1_random` | 0.8876 | 3,760,139 |

### CODE REVIEW

**Comment by stephane-rivaud** [comment]
## Review follow-up: dropout / drop-path homogenization + integration tests

Pushed commit `9b3af59b` with the following changes:

### Module refactor
- **`GrowingDropout`**: Added a concrete `GrowingDropout` class (generic `nn.Dropout` shape) on top of `_GrowingDropoutBase`, alongside the existing `GrowingDropout1d` / `GrowingDropout2d` variants.
- **`growing_drop_path.py`**: Extracted `drop_path()`, `GrowingDropPath`, and `DropPath` alias from `growing_transformer.py` into a shared module (mask-based implementation, `drop_prob` validation).
- **`growing_transformer.py`**: Removed inline `ExtendedDropout`, `DropPath`, and `drop_path`; now imports `GrowingDropout` and `DropPath` from the shared modules.

### Tests
- **`test_growing_dropout.py`**: Coverage for the new `GrowingDropout` class.
- **`test_growing_drop_path.py`**: New unit tests for drop-path behavior.
- **`test_growing_vision_transformer.py`**: Added `TestGrowingTransformerDropPathIntegration` — checks that `extended_forward` matches `forward` after `apply_change` in both **eval** and **train** mode.

### DropPath integration test note
The train-mode equivalence test uses **`dropout=0.0` on the MLP** (while keeping `attention_dropout` and `drop_path_rate` non-zero). This is intentional: `GrowingDropout.extended_forward` applies dropout only to the main branch and leaves `x_ext` unchanged, so with non-zero MLP dropout the merged path after growth is not bitwise identical in train mode. Eval-mode equivalence still holds with dropout enabled. DropPath is unaffected because it runs via `forward` on merged residuals, not via `extended_forward` on split branches.

Happy to discuss whether MLP `extended_forward` should eventually mirror full-tensor dropout for train-mode parity — that would be a separate design change.

**Review by stephane-rivaud — APPROVED** [review]
I made some changes to the PR by refactoring DropPath and Dropout related code.
Tests could be slightly strengthen in a dedicated PR if necessary, but I approve the current PR as is. Tests are failing due to bad luck with random seeds; different fail across different push. This might need to be solved to merge without bypassing the rules.

@TheoRudkiewicz @sylvchev

File: src/gromo/containers/growing_vision_transformer.py [hunk 1]
```diff
@@ -0,0 +1,1217 @@
...
+                    "`in_features` must be a tuple of the form (channels, height, width)."
+                )
+            if pooling not in {None, "cls"}:
+                raise ValueError(
+                    "`GrowingTransformer` mirrors ViTLite and uses cls pooling."
+                )
+            if use_cls_token is False:
+                raise ValueError(
+                    "`GrowingTransformer` mirrors ViTLite and requires a cls token."
+                )
```
Comment (TheoRudkiewicz) [inline src/gromo/containers/growing_vision_transformer.py]: This was never merged before so no need for backward compatibility and legacy code. If it's not too difficult to change it would be good to do it.

File: src/gromo/containers/growing_vision_transformer.py [hunk 2]
```diff
@@ -0,0 +1,1217 @@
...
+                raise ValueError(
+                    "Pass either a positional layer index or `index=...`, not both."
+                )
+            index = scheduling_method
+            scheduling_method = "all"
+        self._growable_layers = _block_growth_targets(self.classifier.blocks)
+        super().set_growing_layers(
+            scheduling_method=scheduling_method,
+            index=index,
+        )
```
Comment (TheoRudkiewicz) [inline src/gromo/containers/growing_vision_transformer.py:650]: Allowing `scheduling_method` to be `str | int` is a bit dangerous. I would advocate to no re-define `set_growing_layers` and keep the parent behavior.

Comment (stephane-rivaud) [reply src/gromo/containers/growing_vision_transformer.py:650]: Resolved. See modifications of the tests in the last commit.

File: src/gromo/containers/growing_vision_transformer.py (hunk 1 above)
Comment (Santiago23florido) [reply src/gromo/containers/growing_vision_transformer.py]: Done, I removed the legacy GrowingTransformer constructor path since it was never merged

**Comment by stephane-rivaud** [comment]
Multi-model + docs review on HEAD `52ebe1af` (Kilo: grok-4.5 / qwen3.7-max / glm-5.2 / minimax-m3 / kilo-auto-efficient; docs-reliability).

**Verdict:** No feature correctness blockers. Theo threads addressed in code (`set_growing_layers` + legacy constructor removal). Legacy review thread resolved as done.

**CI:** tests/ruff/docs green; **pydoclint red** on pre-existing `enumerate_dataloader` DOC404 in `training_utils.py` (unchanged vs main) — out of scope for this PR’s transformer work, but may block merge if the check is required.

**Should-fix (optional, PR-scoped):** (1) strengthen or drop the misleading `evaluate_model` tensor GrowingContainer test; (2) nested outer/classifier `set_growing_layers` scheduling — forward or document outer-only.

**Docs:** whats_new-only is consistent with other containers; not a merge blocker.
